src/main/java/ru/mts/Animals/AbstractAnimal.java

```diff
@@ -1,24 +1,29 @@
 package ru.mts.Animals;
 
 import java.math.BigDecimal;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
 
-public abstract class AbstractAnimal implements Animal
+public abstract class AbstractAnimal implements Animal,Comparable
 {
+    protected LocalDate birthDate  ;
     protected String breed,name,character;
     protected BigDecimal cost;
 
     /**
      * Конструктор AbstractAnimal
      * @param breed порода животного
      * @param name имя животного
+     * @param birthDate дата рождения
      * @param character характер животного
      * @param cost цена животного
      *
      *             cost округляется до 2х знаков после запятой, для округления использован метод Math.round() т.к в т.з предполагается что в параметре cost храниться сразу округленная цена для магазинов
      */
-    public AbstractAnimal(String breed, String name, String character, BigDecimal cost) {
+    public AbstractAnimal(String breed, String name,LocalDate birthDate, String character, BigDecimal cost) {
         this.breed = breed;
         this.name = name;
+        this.birthDate = birthDate;
         this.character = character;
         this.cost = cost.setScale(2, BigDecimal.ROUND_HALF_UP);
```

Comment:
```suggestion
        this.cost = cost.setScale(2, RoundingMode.HALF_UP);
```

idea же показывает что деприкейтед